lib/registry.dart

```diff
@@ -122,7 +122,7 @@ class Registry<T> {
         _timeout = timeout;
 
   _RegistryCache get _cache {
-    _RegistryCache cache = _caches[this];
+    var cache = _caches[this] as _RegistryCache?;
```

Comment:
This entire getter could be
```dart
_RegistryCache get _cache => _caches[this] ??= _RegistryCache();
```
if `_caches` is declared as `Expando<_RegistryCache>`.